controllers/replication/secret_controller.go

```diff
@@ -286,6 +295,9 @@ func (s *SecretSyncController) Reconcile(ctx context.Context, req ctrl.Request)
 }
 
 func requiresUpdate(source, dest client.Object) bool {
+	// TODO: we need to revisit this logic since I think that the change of name (due to adding a prefix) might cause a new UID
```

Comment:
Name itself is not important, but if the source and target have different names, then they're not the same secret, even when replicated inside the same namespace. Thus they would have different UID as they're different objects.

Reply:
That's what I figured, I guess my question is about this line/comment:

```
// In case we target the same cluster
	if source.GetUID() == dest.GetUID() {
```

Can we remove this if we check whether the origin and destination is the same elsewhere?

Reply:
But with a prefix, they're not. They should be replicated. 

What we want to avoid is modifying the same object we read currently. A replicated one, even if located on the same place, is not the same object.

Reply:
That actually makes a good deal of sense now you present it  that way. So you'd like to retain the check on the UIDs (to avoid conflicts where the original tries to overwrite itself) but remove this check to enable a case where a secret is replicating into the same context-namespace but with a different name due to the prefix? I'm ok with this if you can confirm this is what you're thinking.